federation/pkg/kubefed/init/init.go

```diff
@@ -153,19 +159,23 @@ func initFederation(cmdOut io.Writer, config util.AdminConfig, cmd *cobra.Comman
 	cmKubeconfigName := fmt.Sprintf("%s-kubeconfig", cmName)
 
 	// 1. Create a namespace for federation system components
-	_, err = createNamespace(hostClientset, initFlags.FederationSystemNamespace)
+	_, err = createNamespace(hostClientset, initFlags.FederationSystemNamespace, dryRun)
 	if err != nil {
 		return err
 	}
 
 	// 2. Expose a network endpoint for the federation API server
-	svc, err := createService(hostClientset, initFlags.FederationSystemNamespace, serverName)
+	svc, err := createService(hostClientset, initFlags.FederationSystemNamespace, serverName, dryRun)
 	if err != nil {
 		return err
 	}
-	ips, hostnames, err := waitForLoadBalancerAddress(hostClientset, svc)
-	if err != nil {
-		return err
+
+	if !dryRun {
```

Comment:
@irfanurrehman

> The main reason is that in the case of dry run we do want to completely skip this function.

I think you should still push it down and just return there for consistency.

> Are there any specific benefits in declaring and initializing the variable from function return?

Yes. Code conciseness and readability. That's the primary reason we have type inference.